packages/@react-spectrum/table/src/TableView_DEPRECATED.tsx

```diff
@@ -274,13 +274,16 @@ function TableView_DEPRECATED<T extends object>(props: SpectrumTableProps<T>, re
     }
   };
 
-  let isVerticalScrollbarVisible = false;
-  let isHorizontalScrollbarVisible = false;
-  if (bodyRef.current && domRef.current) {
-    // 2 is the width of the border which is not part of the box size
-    isVerticalScrollbarVisible = domRef.current.getBoundingClientRect().width > bodyRef.current.children[0]?.getBoundingClientRect().width + 2;
-    isHorizontalScrollbarVisible = domRef.current.getBoundingClientRect().height > bodyRef.current.children[0]?.getBoundingClientRect().height + 2;
-  }
+  // whenever the viewport changes size, check if scroll bars are visible
+  //
+  let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = useState(false);
+  let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = useState(false);
+  let onVisibleRectChange = useCallback(() => {
+    if (bodyRef.current) {
```

Comment:
Probably should do check that the width changed like you did in the other tableview. But I guess doesn't matter if we aren't releasing it?

Reply:
ah, that was an optimization when i was doing a more expensive calculation. now it's not so bad I don't think, the non-deprecated one should be updated to this one, but yeah, doesn't matter quite yet, I'll put it in my followup tasks